crates/rust-analyzer/src/handlers/notification.rs

```diff
@@ -127,7 +136,7 @@ pub(crate) fn handle_did_close_text_document(
             tracing::error!("orphan DidCloseTextDocument: {}", path);
         }
 
-        if let Some(file_id) = state.vfs.read().0.file_id(&path) {
+        if let Some((file_id, vfs::FileExcluded::No)) = state.vfs.read().0.file_id(&path) {
```

Comment:
I think we want to do this irrespective of exclusion (in the rare case that someone adds an exclusion for a file that has diagnostics, assuming that even works)

Reply:
Exclusions require reload (at least that's what the VSCode client tells me), but sure, why not.

Reply:
Figured as much, though it doesn't hurt to do this either way